reconcile/utils/aws_api.py

```diff
@@ -941,6 +950,16 @@ def get_cluster_nat_gateways_egress_ips(self, account):
 
         return egress_ips
 
+    def get_elasticache_availability_zones(self, account_name: str) -> set[str]:
+        elasticache = self._account_elasticache_client(account_name)
+        subnet_groups = elasticache.describe_cache_subnet_groups()["CacheSubnetGroups"]
```

Comment:
I wouldn't trust AWS API to guarantee this key will always exist. Terraform learned the hard way. :smile:

Reply:
Note `describe_cache_subnet_groups` response is paginated https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/elasticache/client/describe_cache_subnet_groups.html, to ensure get all azs from subnets, we need to implement fetch all using `Marker`.

Reply:
@hemslo, good point! Even though the default records count is set to 100, which probably would cover us for a while, we should do it to future-proof things, indeed. :+1: